### ISSUE TITLE
Waivers for project identifiers

### ISSUE THREAD

**Issue by daveaglick** [licensezero/licensezero-waiver#1]
Given that the Private license grants rights to a particular project identifier (I.e., a version or product line), it would be nice to be able to grant waivers under the same limitations. For example, if I'm granting private licenses based on my major version numbers, I might want to also grant waivers to just the next major version to anyone that licensed the old version within the last 30 days. That waiver should cover the identifier connected to the next major version, not a calendar time-frame.

The other possibility would be to allow granting private licenses directly without requiring a purchase flow. If that option is available this request isn't needed, though I didn't see a capability to do that in the CLI docs.

**Comment by kemitchell** [comment]
@daveaglick neither licensezero.com nor the CLI currently supports issuing private licenses without payment. Nor do developers have direct access to the keys licensezero.com uses to sign such licenses for them.

When you mentioned using the same limitations under a waiver as under the private license, were you referring to the rules under the Sublicensing section? https://licensezero.com/licenses/private

**Comment by daveaglick** [comment]
> When you mentioned using the same limitations under a waiver as under the private license, were you referring to the rules under the Sublicensing section?

I don't think so... The idea is to protect licensees from buying a license for the current major version (which is how I'm planning to provision project identifiers) the day (or month, or several months) before a new major version is released. I'd like to provide a guarantee to licensees that if you buy a license, it'll cover _both_ the major version that you're licensing as well as any _new_ major versions that are released within a given grace period.

I'm fine handing out those additional "upgraded" licenses manually, I just don't see any way to currently accomplish that.

**Comment by daveaglick** [comment]
🤔 after thinking about it a little more, I guess I could accomplish this by automatically granting a time-limited waiver to each licensee at the time they purchase the license. That way they’ll get the perpetual license to that version _plus_ the right to use any version for a given grace period.
